doc/src/guides/solving/solve-ode.md

```diff
@@ -0,0 +1,262 @@
+>>> solutions = dsolve(eqs); solutions
+[[Eq(y(x), C1 - C2*exp(x)), Eq(z(x), C2*exp(x))], [Eq(y(x), C1 + C2*exp(x)), Eq(z(x), C2*exp(x))]]
+>>> solutions_list = [] # nested list approach
+>>> for solution in solutions:
+...     solution_dict = {}
+...     for fn in solution:
+...             solution_dict.update({fn.lhs: fn.rhs})
+...     solutions_list.append(solution_dict)
+>>> solutions_list
+[{y(x): C1 - C2*exp(x), z(x): C2*exp(x)}, {y(x): C1 + C2*exp(x), z(x): C2*exp(x)}]
+>>> solutions_list = [{fn.lhs:fn.rhs for fn in solution} for solution in solutions]
+>>> solutions_list # nested comprehension approach
+[{y(x): C1 - C2*exp(x), z(x): C2*exp(x)}, {y(x): C1 + C2*exp(x), z(x): C2*exp(x)}]
+>>> solutions_list[0][y(x)]
+C1 - C2*exp(x)
+```
+
+### Work With Arbitrary Constants *include such a section?*
+
+## Ordinary Differential Equation Solving Hints
+
+### Return Unevaluated Integrals
+
+By default, {func}`~.dsolve` attempts to evaluate the integrals it produces to
+solve your ordinary differential equation. You can disable evaluation of the
+integrals by using {ref}`hints` ending with `_Integral`, for example
+`separable_Integral`. This is useful because
+{func}`~sympy.core.expr.Expr.integrate` is an expensive routine. SymPy may hang
+(appear to never complete the operation) because of a difficult or impossible
+integral, so using an `_Integral` hint will at least return an (unintegrated)
+result, which you can then consider. The simplest way to disable integration is
+with the `all_Integral` hint because you do not need to know which hint to
+supply: for any hint with a corresponding `_Integral` hint, `all_Integral` only
+returns the `_Integral` hint.
+
+### Select a Specific Solver
+
+You may wish to select a specific solver using a hint for a couple of reasons:
+- educational purposes: for example if you are learning about a specific method
+  to solve ODEs and want to get a result that exactly matches that method
+- form of the result: sometimes an ODE can be solved by many different solvers,
+  and they can return different results. They will be mathematically equivalent,
+  though the arbitrary constants may not be. {func}`~.dsolve` by default tries
+  to use the "best" solvers first, which are most likely to return the most
+  usable output, but it is not a perfect heuristic. For example, the "best"
+  solver may produce a result with an integral that SymPy cannot solve, but
+  another solver may produce a different integral that SymPy can solve. So if
+  the solution isn't in a form you like, you can try other hints to check
+  whether they give a preferable result.
+
+## Not All Equations Can Be Solved
+
+### Equations With No Solution
+
+*Equations with no solution content*
+
+### Equations With No Analytical Solution
+
+*Equations with no analytical solution content*
+
+### Equations Which Have An Analytical Solution, and SymPy Cannot Solve
+
+*Equations which have an analytical solution, and SymPy cannot solve content*
```

Comment:
Can someone (perhaps @oscarbenjamin) come up with an ODE (one or system) which has a closed-form solution, but which SymPy cannot solve?

Reply:
Any example like that will always be fragile because in principle a future version of SymPy will be able to solve it. In fact putting it in the guide makes it particularly likely that `dsolve` would be improved in future to handle that case.